desktop/libs/hadoop/src/hadoop/fs/webhdfs.py

```diff
@@ -896,6 +931,27 @@ def _get_redirect_url(self, webhdfs_ex):
       LOG.exception("Failed to read redirect from response: %s (%s)" % (webhdfs_ex, ex))
       raise webhdfs_ex
 
+
+  def _set_params_with_delegation_token(self, params):
+    try:
+      token = self.get_delegation_token(self.user)
+    except:
+      with self.get_delegation_token.cache_lock:
+        self.get_delegation_token.cache.pop(self.get_delegation_token.cache_key(self.user), None)
+      token = self.get_delegation_token(self.user)
+    if token:
+      params['delegation'] = token
+      # doas should not be present with delegation token as the token includes the username
+      # https://hadoop.apache.org/docs/r1.0.4/webhdfs.html
+      if 'doas' in params:
+        del params['doas']
+      if 'user.name' in params:
+        del params['user.name']
+    return params
+
+  # a renew-interval (default is 24 hours)
+  # https://blog.cloudera.com/hadoop-delegation-tokens-explained/
+  @ttl_cache(maxsize=128, ttl=timedelta(hours=24), timer=datetime.now, typed=False)
```

Comment:
@ranade1 can you review the use of `cachetools` here and inclusion into requirements above?